core/src/net/mod.rs

```diff
@@ -1263,18 +1301,35 @@ pub mod net_test_helpers {
                 blocked_ip: Vec::new(),
                 max_channels_reached: 0,
                 network_out_of_buffers: 0,
+                network_status_queue_sender: None,
+                started_promise: None,
             }
         }
 
+        /// Sets a sender which the NetworkStatusCounter will foward all the NetworkStatus events to
+        pub fn set_status_sender(&mut self, sender: Sender<NetworkStatus>) {
+            self.network_status_queue_sender = Some(sender);
+        }
+
         /// triggers the given `request` on the NetworkStatusPort
         pub fn send_status_request(&mut self, request: NetworkStatusRequest) {
             debug!(self.ctx.log(), "Sending Status Request. {:?}", request);
             self.network_status_port.trigger(request);
         }
+
+        /// Returns a promise that will be fulfilled when the component starts
+        pub fn get_started_future(&mut self) -> KFuture<()> {
```

Comment:
Just `started_future` for consistency with other methods. Also this "creates" something it doesn't "get" something.